File: src/python/pdptwse.py
import sys
from modules.parameters import read_input_parameters
from modules.data import read_data
from modules.formulations import (
    melo_mip_formulation,
    melo_hexaly_formulation,
)
from modules.solutions import print_detail_melo_formulation_solution, validate_solution
from gurobipy import Env, Model


# Initialize Gurobi environment
try:
    GRB_ENV = Env()
except Exception as _:
    GRB_ENV = None  # or a dummy object

if GRB_ENV is not None:
    model = Model(env=GRB_ENV)

# Read the parameters from command line
params = read_input_parameters(sys.argv)

# # Read instance data
inst = read_data(params)

sol = None
if params.method_type == "form":
    if params.method_code == "melo":
        sol = melo_mip_formulation(GRB_ENV, inst, params)
    if params.method_code == "melo_hx":
        sol = melo_hexaly_formulation(inst, params)

if sol is not None and params.print_sol == 1:
    print_detail_melo_formulation_solution(inst, sol)
    # if validate_solution(inst, sol, params):
    if validate_solution(inst, sol, params):
        print("Everything is awesome!")
    else:
        print("Infeasible solution :(")

File: src/julia/pdptwse.jl
push!(LOAD_PATH, "modules/")

using Data
using Parameters
using Formulations
using Gurobi
using JuMP
using GreedyHeuristicMutate
using Multistart
using Solutions

# Check if Gurobi is available
const GRB_ENV = let
    try
        @eval using Gurobi
        Gurobi.Env()
    catch e
        nothing  # or a dummy object
    end
end
if GRB_ENV !== nothing
	Model(() -> Gurobi.Optimizer(GRB_ENV))
	# Model(Gurobi.Optimizer)
end

# Read the parameters from command line
params = readInputParameters(ARGS)

# Read instance data
inst = read_data(params)

# Solve the problem according to the selected method
sol::Union{Nothing, Solution} = nothing
if params.methodType == "heur"
	if params.methodCode == "greedy"
		sol = GreedyHeuristicMutate.greedyHeuristicMutate(inst, params)
	elseif params.methodCode == "mslp"
		Multistart.multistartlpinitialsetup(GRB_ENV, inst, params)
		sol = Multistart.multistartlp(GRB_ENV, inst, params)
	end
end

# Print solution details and validate solution
if sol !== nothing
	if params.printsol == 1
		printDetailMeloFormulationSolution(inst, sol)
	end
	
	if validate_solution(inst, sol, params)
		println("Feasible solution! :D")
	else
		println("Infeasible solution! :(")
	end
end